### PR TITLE
Task/trial1 create update delete subs

### PR DESCRIPTION
**Pull request by kzangeli**
Rest of new sub-cache implemented (but not tested):

The sub-cache is now modified when a subscription is:
- created
- updated
- deleted

### CODE REVIEW

File: src/lib/mongoBackend/mongoSubCache.cpp [hunk 1]
```diff
@@ -481,19 +441,73 @@ void mongoSubCacheDestroy(void)
...
   //
-  BSONElement  idField = subP->getField("_id");
+  BSONElement  idField = sub.getField("_id");
 
   if (idField.eoo() == true)
   {
-    LM_E(("Database Error (error retrieving _id field in doc: '%s')", subP->toString().c_str()));
-    return;
+    LM_E(("Database Error (error retrieving _id field in doc: '%s')", sub.toString().c_str()));
+    return -1;
```
Comment (fgalan) [inline src/lib/mongoBackend/mongoSubCache.cpp]: Maybe a dictionary of codes that the fucntion may return should be added to the function comment preamble.

Comment (kzangeli) [inline src/lib/mongoBackend/mongoSubCache.cpp]: Sure, that is always good

Comment (rbl219) [inline src/lib/mongoBackend/mongoSubCache.cpp]: LGTM

Comment (kzangeli) [inline src/lib/mongoBackend/mongoSubCache.cpp]: Fixed in 8ac8f45 :-)

**Comment by fgalan** [comment]
LGTM

**Pull request merged**